prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -237,25 +237,42 @@ class ReportFunction {
             "${service.organization.description} (${service.name})"
         else
             service.organization.description
-        val event = when {
+        val loggerMsg: String
+        when {
             validatedRequest.options == Options.SkipSend -> {
-                ReportEvent(Event.EventAction.NONE, report.id)
+                val event = ReportEvent(Event.EventAction.NONE, report.id)
+                workflowEngine.dispatchReport(event, report, txn)
+                actionHistory.trackCreatedReport(event, report, service)
+                loggerMsg = "Queue: ${event.toQueueMessage()}"
             }
             service.batch != null -> {
                 val time = service.batch.nextBatchTime()
-                val destination = "Sending ${report.itemCount} items to $serviceDescription at $time"
-                destinations += destination
-                ReceiverEvent(Event.EventAction.BATCH, service.fullName, time)
+                destinations += "Sending ${report.itemCount} items to $serviceDescription at $time"
+                val event = ReceiverEvent(Event.EventAction.BATCH, service.fullName, time)
+                workflowEngine.dispatchReport(event, report, txn)
+                actionHistory.trackCreatedReport(event, report, service)
+                loggerMsg = "Queue: ${event.toQueueMessage()}"
```

Comment:
This is a bug that needs fixing before you merge this PR.   Down inside dispatchReport, format is secretly changing to CSV, but the format is now incorrect in ActionHistory.   I suggest the format decision should be refactored out and made here, then sent to both dispatchReport and trackCreatedReport.

Again, note that this is temporary.  My intention is that the new `report_file `table will be the primary table for all report information, not the `task` table.   The `task` table will remain, but only as a mutable "todo list" of work that needs to be done, with the details about reports properly normalized into `report_file`. Right now, `task` and `report_file` are redundant, which is a dangerous situation I intend to fix.

Furthermore, see my comment on putting Send on a cron timer.     That would enable you to skip the batch step entirely, and maybe you don't even need the `forceFormat` stuff all?

Reply:
Ok. I'll look at this bug.

Reply:
Actually, I think you should go ahead and merge, and I'll fix this.  I'd like to do my merge on top of yours, as I know there are going to be a lot of conflicts.